public/app/features/datasources/state/actions.ts

```diff
@@ -276,8 +276,8 @@ export function updateDataSource(dataSource: DataSourceSettings) {
       const formattedError = parseHealthCheckError(err);
 
       dispatch(testDataSourceFailed(formattedError));
-
-      return Promise.reject(err);
+      const errorInfo = isFetchError(err) ? err.data : { message: 'An unexpected error occurred.', traceId: '' };
```

Comment:
I am wondering if we should "swallow" the error details in case it's not a `FetchError` (as it's most probably the fetch error which is [containing the config options](https://github.com/grafana/grafana/blob/main/public/app/core/services/backend_srv.ts#L356)). Also, would we need the `traceId` [from the FetchError](https://github.com/grafana/grafana/blob/1d6d5cc2cd1fada191db82f7f045851e10908190/packages/grafana-runtime/src/services/backendSrv.ts#L126)? (It looks to me that under `data` we have `traceID`).